### PR TITLE
Add ibc support

### LINKED ISSUE #636: Add ibc messages to codec
I was trying to use stargate.codec to create IBC client messages, but I couldn't find the Msg types.

Looking at [the generation scripts](https://github.com/cosmos/cosmjs/blob/6182791cb82eb48466a6410ac970e5d37ba0f808/packages/stargate/scripts/predefine-proto.sh#L42-L47), I saw that `tx.proto` and in 2 cases `query.proto` were not compiles, although they are present in those directories: https://github.com/cosmos/cosmos-sdk/tree/master/proto/ibc/core/client/v1

Please add those generated codecs, so I can try to create messages.

Also, if it is not too much work, a query extension for client and connections would also be useful (although I guess I could do that outside of the package if the protobuf definitions are there?)

### PR DESCRIPTION
**Pull request by willclarktech**
Part 1 of #636

### CODE REVIEW

**Review by webmaster128 — COMMENTED** [review]
Hmm, is this the way to go? Should @cosmjs/stargate expose codecs it does not use internally?

This might be friendly for starters, but sooner or later users will reach the point where they need https://github.com/cosmos/cosmjs/issues/640.

**Comment by webmaster128** [comment]
What's your experience building the IBC relayer?

**Comment by willclarktech** [comment]
In `ts-relayer` we just have our own codec. This was straightforward for us to do because we already knew how to do it from CosmJS, but it's probably a real pain for newcomers. I have no opinions about what should/shouldn't be exposed here because I'm not very clear on who we're targeting with this exposed codec and what their use cases are (besides us).

**Comment by ethanfrey** [comment]
I think we definitely need #640 as people will have app-specific proto. I am now happy to have the ibc in the relayer as we can iterate on it quickly when I need to adjust it (which happened often).

That said, queries that are expected to be commonly used should be in here and exposed in the StargateClient. or SigningStargateClient. I would be happy if someone took over https://github.com/cosmos/cosmjs/pull/690 (which I paused as added a new message was quite hard with the current setup).

Adding support for the ibc-transfer queries and messages seems very useful, so wallets can easily send a token over ics20 and query the denom behind a hash (you get something like "ibc/E957AD87675BEA6576576" as denom, which needs the DenomTrace query to decypher.

File: packages/stargate/src/queries/ibc.ts [hunk 1]
```diff
@@ -14,76 +17,155 @@ import {
...
-        portId: string,
-        channelId: string,
-        sequence: number,
-      ) => Promise<QueryPacketCommitmentResponse>;
-      readonly packetCommitments: (
-        portId: string,
-        channelId: string,
-        paginationKey?: Uint8Array,
-      ) => Promise<QueryPacketCommitmentsResponse>;
+    readonly channel: {
```
Comment (willclarktech) [inline packages/stargate/src/queries/ibc.ts:66]: Do we need any more verified queries at this time?

File: packages/stargate/src/queries/ibc.spec.ts [hunk 2]
```diff
@@ -14,220 +14,562 @@ async function makeClientWithIbc(rpcUrl: string): Promise<[QueryClient & IbcExte
...
+        xit("works", async () => {
+          pendingWithoutSimapp();
+          const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+          const response = await client.ibc.unverified.channel.consensusState(
+            ibcTest.portId,
+            ibcTest.channelId,
+            // TODO: Find valid values
+            0,
+            0,
```
Comment (willclarktech) [inline packages/stargate/src/queries/ibc.spec.ts:111]: I'm not sure which values I need here. Do I need to define them dynamically? @ethanfrey

File: packages/stargate/src/queries/ibc.spec.ts [hunk 3]
```diff
@@ -14,220 +14,562 @@ async function makeClientWithIbc(rpcUrl: string): Promise<[QueryClient & IbcExte
...
+      });
+
+      describe("stateTm", () => {
+        it("works", async () => {
+          pendingWithoutSimapp();
+          const [client, tmClient] = await makeClientWithIbc(simapp.tendermintUrl);
+
+          const response = await client.ibc.unverified.client.stateTm(ibcTest.clientId);
+          expect(response.chainId).toEqual("ibc-1");
+          // TODO: Fill these expectations out
```
Comment (willclarktech) [inline packages/stargate/src/queries/ibc.spec.ts:429]: This result has a lot of properties. Do we need to assert against all of them?

Comment (webmaster128) [reply packages/stargate/src/queries/ibc.spec.ts:429]: I think it is fine like this. Good to have the test in place. When someone is working on this particular call or we run into issues, we can fill out the test body more and more.

**Review by webmaster128 — APPROVED** [review]

File: packages/stargate/src/queries/ibc.ts (hunk 1 above)
Comment (willclarktech) [reply packages/stargate/src/queries/ibc.ts:66]: See discussion here: https://github.com/cosmos/cosmjs/issues/713

**Pull request merged**